NGitLab.Tests/SnippetsTest.cs

```diff
@@ -75,8 +75,45 @@ public async Task Test_snippet_inProject(VisibilityLevel visibility)
             snippetClient.Delete(newSnippet.ProjectId, returnedProjectSnippet.Id);
         }
 
+        [TestCase(VisibilityLevel.Private)]
+        [TestCase(VisibilityLevel.Internal)]
+        [TestCase(VisibilityLevel.Public)]
+        [NGitLabRetry]
+        public async Task Test_snippet_files(VisibilityLevel visibility)
+        {
+            using var context = await GitLabTestContext.CreateAsync();
+            var project = context.CreateProject();
+            var snippetClient = context.Client.Snippets;
+
+            var guid = Guid.NewGuid().ToString("N");
+            var snippetName = "testSnip" + guid;
+
+            // arrange
+            var newSnippet1 = new SnippetCreate
+            {
+                Title = snippetName,
+                Visibility = VisibilityLevel.Public,
+                Files = new[]
+                {
+                    new SnippetCreateFile() { Action = SnippetCreateFile.ActionType.Create, FilePath = "Path1.txt", Content = "Content1" },
+                    new SnippetCreateFile {  Action = SnippetCreateFile.ActionType.Create, FilePath = "Path2.txt", Content = "Content2" },
+                },
+            };
+
+            // act - assert
+            snippetClient.Create(newSnippet1);
+            var snippet = snippetClient.All.First();
+
+            Assert.IsNotNull(snippet.Files);
+            Assert.IsNotEmpty(snippet.Files);
+            Assert.IsTrue(snippet.Files.Length == 2);
```

Comment:
To get better error message, you can use `Assert.AreEqual`

```suggestion
            Assert.AreEqual(2, snippet.Files.Length);
```